k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -3022,7 +2995,8 @@ public void onMessageBuildCancel() {
     @Override
     public void onMessageBuildException(MessagingException me) {
         Log.e(K9.LOG_TAG, "Error sending message", me);
-        Toast.makeText(MessageCompose.this, getString(R.string.send_aborted), Toast.LENGTH_LONG).show();
+        Toast.makeText(MessageCompose.this,
+                getString(R.string.send_failed_reason, me.getLocalizedMessage()), Toast.LENGTH_LONG).show();
         currentMessageBuilder = null;
```

Comment:
I did this because otherwise messages like "Signing key not found" would not get through to the user, giving them no clue whatsoever what was wrong other than "Send aborted"...